base/float.jl

```diff
@@ -503,7 +503,7 @@ isfinite(x::Float16) = reinterpret(UInt16,x)&0x7c00 != 0x7c00
 isfinite(x::Real) = decompose(x)[3] != 0
 isfinite(x::Integer) = true
 
-isinf(x::Real) = !isnan(x) & !isfinite(x)
+isinf(x::Real) = (reinterpret(Unsigned, x) & ~sign_mask(T)) == exponent_mask(T)
```

Comment:
Perhaps we also need an extra `IEEEFloat` abstract type, for which these things are valid?